UnityWeld_Editor/AnimatorParameterBindingEditor.cs

```diff
@@ -0,0 +1,337 @@
+        )
+        {
+            var animator = targetScript.GetComponent<Animator>();
+
+            AnimatorControllerParameter[] properties = new AnimatorControllerParameter[] { };
+
+            // For whatever reason, accessing Animator.parameters while the GameObject the
+            // Animator is on is inactive causes it to return an empty array, aswell as fill
+            // the unity console with warnings. This behaviour is undocumented, and this is just
+            // a work around which seems to work fine
+            if (!animator.gameObject.activeInHierarchy)
+            {
+                UnityEditorInternal.ComponentUtility.CopyComponent(animator.transform);
+                var parent = animator.transform.parent;
+                var siblingIndex = animator.transform.GetSiblingIndex();
+                var isActive = animator.gameObject.activeSelf;
+                animator.transform.SetParent(null);
+                animator.gameObject.SetActive(true);
+                properties = animator.parameters;
+                animator.transform.SetParent(parent);
+                animator.transform.SetSiblingIndex(siblingIndex);
+                animator.gameObject.SetActive(isActive);
+                UnityEditorInternal.ComponentUtility.PasteComponentValues(animator.transform);
+            }
+            else
+            {
+                properties = animator.parameters;
+                //Another odd fix to refresh the parameters
+                if (properties == null || properties.Length == 0)
+                {
+                    var isActive = animator.gameObject.activeSelf;
+                    animator.gameObject.SetActive(false);
+                    animator.gameObject.SetActive(true);
+                    animator.gameObject.SetActive(isActive);
+                    properties = animator.parameters;
+                }
+            }
+
+            var propertyNamesAndTypes = properties
+                .Select(m => new AnimatorParameterTypeAndName(m.name, m.type))
+                .ToArray();
+            var selectedIndex = Array.IndexOf(propertyNamesAndTypes, curPropertyValue);
+            var content = properties.Select(prop => new GUIContent(string.Concat(
+                    prop.name,
+                    " : ",
+                    prop.type.ToString()
+                )))
+                .ToArray();
+
+            var newSelectedIndex = EditorGUILayout.Popup(label, selectedIndex, content);
+            if (newSelectedIndex != selectedIndex)
+            {
+                var newSelectedProperty = properties[newSelectedIndex];
+
+                UpdateProperty(
+                    propertyValueSetter,
+                    curPropertyValue,
+                    new AnimatorParameterTypeAndName(newSelectedProperty.name, newSelectedProperty.type),
+                    "Set Animator parameter"
+                );
+
+                switch (newSelectedProperty.type)
```

Comment:
See commend above about style for switch statements. I also noticed this is an exact duplicate of the code at line 244 - would be good if this was refactored out into a method that takes an `AnimatorControllerParameterType` and returns a `Type` to get rid of the duplication.